java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoReportMerger.java

```diff
@@ -64,23 +63,23 @@ public static void mergeReports(File reportOverall, File... reports) {
       dataStore.accept((IExecutionDataVisitor) visitor);
     } catch (IOException e) {
       throw new SonarException(String.format("Unable to write overall coverage report %s", reportOverall.getAbsolutePath()), e);
-    } finally {
-      Closeables.closeQuietly(outputStream);
     }
   }
 
   private static boolean loadSourceFiles(ISessionInfoVisitor infoStore, IExecutionDataVisitor dataStore, File... reports) {
-    boolean isCurrentVersionFormat = true;
+    Boolean isCurrentVersionFormat = null;
     for (File report : reports) {
       if (report.isFile()) {
-        try {
-          isCurrentVersionFormat = AbstractAnalyzer.readJacocoReport(report, dataStore, infoStore);
-        } catch (IOException e) {
-          throw new SonarException(String.format("Unable to read %s", report.getAbsolutePath()), e);
+        JacocoReportReader jacocoReportReader = new JacocoReportReader(report).readJacocoReport(dataStore, infoStore);
+        boolean reportFormatIsCurrent = jacocoReportReader.useCurrentBinaryFormat();
+        if (isCurrentVersionFormat == null) {
+          isCurrentVersionFormat = reportFormatIsCurrent;
+        } else if (isCurrentVersionFormat != reportFormatIsCurrent) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)